### PR TITLE
`.ci/write-dockerfile.sh`: Move here from build/bin/

### PR DESCRIPTION
**Pull request by mkoeppe**
... as suggested by @kwankyu in https://groups.google.com/g/sage-devel/c/dEa3i2Fn3ZY/m/nzUr2wc_AAAJ

To test: Commands such as `tox -e docker-ubuntu-jammy-minimal -- config.status` should still work.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -183,8 +183,10 @@ jobs:
           sudo apt-get clean
           df -h
       - name: Update Sage packages from upstream artifact
+        # Handle both the old and new location of write-dockerfile.sh,
+        # because docker.yml is a reusable workflow.
         run: |
```
Comment (kwankyu) [inline .github/workflows/docker.yml:188]: Sorry, but would you explain this: "because docker.yml is a reusable workflow"?

Comment (kwankyu) [reply .github/workflows/docker.yml:188]: I think I know what "a reusable workflow" is, but why need to handle both the old and new location?

**Comment by kwankyu** [comment]
> To test: Commands such as `tox -e docker-ubuntu-jammy-minimal -- config.status` should still work.

Yes, this works well.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (mkoeppe) [reply .github/workflows/docker.yml:188]: Reusable = invoked by other workflows, which includes workflows in other repositories, see links in 
- https://github.com/sagemath/sage/issues/33338

Comment (mkoeppe) [reply .github/workflows/docker.yml:188]: I cannot simultaneously update all of these workflows, so our reusable workflow needs to support both versions until such updates are completed.

Comment (kwankyu) [reply .github/workflows/docker.yml:188]: I see. 

The upstream repos use the reusable workflow, `.github/workflows/docker.yml`. But they do not directly use `write-dockerfile.sh`, which is the only file moved to the new location. Right?

Then, as long as the location of  `.github/workflows/docker.yml` doesn't change, the new location of `write-dockerfile.sh` doesn't matter for the upstream repos. Does it?

Comment (mkoeppe) [reply .github/workflows/docker.yml:188]: That's basically true, but the URL of the workflow and the branch of the repository that it's using are specified separately, see for example:
- https://github.com/scipopt/SCIP-SDP/blob/main/.github/workflows/ci-sage.yml#L87
- https://github.com/scipopt/SCIP-SDP/blob/main/.github/workflows/ci-sage.yml#L97

They may be out of sync for some of the workflows.

Comment (kwankyu) [reply .github/workflows/docker.yml:188]: This business is more complicated in practice than it should be...

OK. Thanks for the explanation.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks for the review!

**Pull request merged**